Client/Ecosia/Analytics/AnalyticsNotificationSettings.swift

```diff
@@ -0,0 +1,53 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UserNotifications
+
+/// A protocol that abstracts the `UNUserNotificationCenter` to allow for dependency injection and mocking.
+/// Utilized for integration tests. Specifically `AnalyticsSpyTests`
+///
+/// By conforming to this protocol, you can replace the notification center with a mock implementation during testing.
+protocol AnalyticsUserNotificationCenterProtocol {
+    /// Retrieves the notification settings asynchronously.
+    ///
+    /// - Parameter completionHandler: A closure that is executed with the retrieved `AnalyticsUNNotificationSettingsProtocol` object.
+    func getNotificationSettingsProtocol(completionHandler: @escaping (AnalyticsUNNotificationSettingsProtocol) -> Void)
```

Comment:
Have you considered using `async await` instead of the handler pattern? Does anything speak against it? If applicable I'd favour the more modern syntax

Reply:
Thanks for the review @ecotopian !
Of course, I'd have favored the `async await` approach first, but that would have brought some changes at a higher level (where the function calling this one is invoked) like making the whole upper level code conform to the `async await` pattern. Hence I decided to keep the code as is to leave the upper level one untouched.

Reply:
I see - yeah then let's play it safe and avoid the domino effects of introducing async await here